src/classes/devices/common/util.ts

```diff
@@ -0,0 +1,48 @@
+import { IOSAdvertisingData } from '../ble.types';
+
+export function to128bitUUID(uuid: string) {
+  // nothing to do
+  switch (uuid.length) {
+    case 4:
+      return `0000${uuid.toUpperCase()}-0000-1000-8000-00805F9B34FB`;
+    case 8:
+      return `${uuid.toUpperCase()}-0000-1000-8000-00805F9B34FB`;
+    case 36:
+      return uuid.toUpperCase();
+    default:
+      throw new Error('invalid uuid: ' + uuid);
+  }
+}
+
+export function parseAdvertisingManufacturerData(
+  adv: ArrayBuffer | IOSAdvertisingData
+): Uint8Array | null {
+  if (
+    adv &&
+    'kCBAdvDataManufacturerData' in adv &&
+    adv.kCBAdvDataManufacturerData instanceof ArrayBuffer
+  ) {
+    return new Uint8Array(adv.kCBAdvDataManufacturerData);
+  }
+
+  if (adv instanceof ArrayBuffer) {
+    const view = new Uint8Array(adv);
+    if (!view.length) {
+      return null;
+    }
+
+    for (let i = 0; i < view.length - 1; ) {
```

Comment:
Is this realy working?

Reply:
What do you mean? Yes, i tested it on Android and it works.

Reply:
I can add some tests for it

Reply:
Alright. The code looked like something missing.
Specially the I++ made me curious

Reply:
Therefore good to go. I gonna merge